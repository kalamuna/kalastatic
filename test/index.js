'use strict'

const execFile = require('child_process').execFile
const exec = require('child_process').exec
const fs = require('fs')
const path = require('path')
const assertDir = require('assert-dir-equal')
const KalaStatic = require('..')
const test = require('testit')
const nconf = require('nconf')
const extend = require('extend-shallow')
const rimraf = require('rimraf')

function setupTest(name, opts) {
  test(name, () => {
    return new Promise((resolve, reject) => {
      // Create the configuration for the test.
      const conf = new nconf.Provider()

      // Force the settings for the test.
      const testOpts = {
        base: path.join('test', 'fixtures', name)
      }
      extend(testOpts, opts)
      conf.overrides(testOpts)

      // Create the environment.
      const kalastatic = new KalaStatic(conf)
      kalastatic.build().then(() => {
        // Make sure the build passes.
        const base = kalastatic.nconf.get('base')
        const build = path.join(base, kalastatic.nconf.get('destination'))
        const expected = path.join(base, 'expected')
        assertDir(build, expected)

        // Continue the test suite.
        resolve()
      }, reject).catch(reject)
    })
  })
}

setupTest('basic')
setupTest('layouts')
setupTest('styles', {
  kss: true
})
setupTest('twig-filters', {
  pluginOpts: {
    'metalsmith-jstransformer': {
      engineLocals: {
        twig: {
          awesomeSauce: 'This Is The AwesomeSauce!'
        }
      }
    }
  }
})

setupTest('namespaces', {
  pluginOpts: {
    'metalsmith-jstransformer': {
      engineOptions: {
        twig: {
          namespaces: {
            icons: '/images/icons/'
          }
        }
      }
    }
  }
})

test('cli', done => {
  const options = {
    cwd: 'test/fixtures/basic'
  }
  execFile(fs.realpathSync('bin/kalastatic'), [], options, (err, stdout, stderr) => {
    if (err) {
      return done(err)
    }
    if (stderr) {
      return done(stderr.toString())
    }
    assertDir('test/fixtures/basic/build', 'test/fixtures/basic/expected')
    done()
  })
})

test('createcomponent', done => {
  const options = {
    cwd: '.'
  }
  rimraf.sync('test/fixtures/createcomponent/build/components')
  exec('bin/kalastatic cc --directory=test/fixtures/createcomponent/build/components atoms/links', options, (err, stdout, stderr) => {
    if (err) {
      return done(err)
    }
    if (stderr) {
      return done(stderr.toString())
    }
    assertDir('test/fixtures/createcomponent/build', 'test/fixtures/createcomponent/expected')
    done()
  })
})
