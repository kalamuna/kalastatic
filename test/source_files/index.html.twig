{% set title = "Hello World!" %}

<!DOCTYPE html>
<html>
  <head>
    <title>{{ title }}</title>
    {{ kalastatic_stylesheets }}
  </head>
  <body>
    {% include '@components/body.twig' %}

    {{ attach_library('samplelibrary') }}

    {{ kalastatic_scripts }}
  </body>
</html>
