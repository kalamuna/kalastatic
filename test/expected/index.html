
<!DOCTYPE html>
<html>
  <head>
    <title>Hello World!</title>
    <link href="/styles.css" rel="stylesheet">
  </head>
  <body>
    <h1>Hello World!</h1>
<img src="assets/image.png" />

    

    <script src="/sampleJavaScript.js" ></script>
  </body>
</html>
